grin/src/Grin/Grin.hs

```diff
@@ -124,7 +125,7 @@ showTS :: Show a => a -> Name
 showTS = packName . show
 
 concatPrograms :: [Program] -> Program
-concatPrograms prgs = Program (concat exts) (concat defs) where
+concatPrograms prgs = Program (nub $ concat exts) (concat defs) where
```

Comment:
I am not sure, if the `nub` is the right choice here. As we have the assumption that exts will be unique, if not we should throw an exception letting the clients know that they behave badly. 

What was the usecase when we created a duplicated ext joining programs?

Reply:
If the program already contains external definitions and we also add the primops prelude "manually" through the pipeline, the externals get duplicated. That's why I introduced this extra `nub` here.

Reply:
@csabahruska What do you think?

Reply:
Than I am also ok with the `nub`